cdap-ui/app/cdap/components/FieldLevelLineage/v2/Context/FllContext.tsx

```diff
@@ -140,6 +159,54 @@ export class Provider extends React.Component<{ children }, IContextState> {
     });
   };
 
+  private updateLineageFromRange(selection: string, start: number | string, end: number | string) {
+    const newState = {
+      selection,
+      start: null,
+      end: null,
+    };
+    // start and end are only set for custom date range
+    if (selection === TIME_OPTIONS[0]) {
+      newState.start = start;
+      newState.end = end;
+    }
+
+    this.setState(newState, () => {
+      const namespace = getCurrentNamespace();
+      const qParams = parseQueryString();
+      const timeParams: ITimeParams = {
+        selection,
+        range: { start, end },
+      };
+      fetchFieldLineage(this, namespace, this.state.target, qParams, timeParams);
```

Comment:
No!! Do not pass on the context `this` around to utility function. This is dangerous.

Reply:
Refactored to avoid passing in Context component -fetchFieldLineage got moved to FllContext, which calls getFieldLineage from FllContextHelper.